src/groups/group.jl

```diff
@@ -289,69 +223,81 @@ Identity element $e ∈ \mathcal{G}$, such that for any element $p ∈ \mathcal{
 $p \circ e = e \circ p = p$.
 The returned element is of a similar type to `p`.
 """
-identity(M::Manifold, p) = identity(M, p, is_decorator_manifold(M))
-identity(M::Manifold, p, ::Val{true}) = identity(M.manifold, p)
-function identity(M::Manifold, p, ::Val{false})
-    return error("identity not implemented on $(typeof(M)) for points $(typeof(p))")
-end
+identity(::AbstractGroupManifold, ::Any)
+@decorator_transparent_signature identity(G::AbstractDecoratorManifold, p)
 function identity(G::AbstractGroupManifold, p)
     y = allocate_result(G, identity, p)
     return identity!(G, y, p)
 end
 
-identity!(M::Manifold, q, p) = identity!(M, q, p, is_decorator_manifold(M))
-identity!(M::Manifold, q, p, ::Val{true}) = identity!(M.manifold, q, p)
-function identity!(M::Manifold, q, p, ::Val{false})
-    return error("identity! not implemented on $(typeof(M)) for points $(typeof(q)) and $(typeof(p))")
+@decorator_transparent_signature identity!(G::AbstractDecoratorManifold, q, p)
+function decorator_transparent_dispatch(
+    ::typeof(identity!),
+    ::AbstractGroupManifold,
+    q,
+    p,
+)
+    return Val(:intransparent)
 end
 
-isapprox(M::Manifold, p, e::Identity; kwargs...) = isapprox(M, e, p; kwargs...)
-function isapprox(M::Manifold, e::Identity, p; kwargs...)
-    is_decorator_group(M) === Val(true) && return isapprox(base_group(M), e, p; kwargs...)
-    error("isapprox not implemented for manifold $(typeof(M)) and points $(typeof(e)) and $(typeof(p))")
-end
-function isapprox(M::Manifold, e::E, ::E; kwargs...) where {E<:Identity}
-    is_decorator_group(M) === Val(true) && return isapprox(base_group(M), e, e; kwargs...)
-    error("isapprox not implemented for manifold $(typeof(M)) and points $(typeof(e)) and $(typeof(e))")
-end
 function isapprox(G::GT, e::Identity{GT}, p; kwargs...) where {GT<:AbstractGroupManifold}
     return isapprox(G, identity(G, p), p; kwargs...)
 end
-function isapprox(
-    ::GT,
-    ::E,
-    ::E;
-    kwargs...,
-) where {GT<:AbstractGroupManifold,E<:Identity{GT}}
-    return true
-end
-function isapprox(G::GT, p, e::Identity{GT}; kwargs...) where {GT<:GroupManifold}
+function isapprox(G::GT, p, e::Identity{GT}; kwargs...) where {GT<:AbstractGroupManifold}
     return isapprox(G, e, p; kwargs...)
 end
-function isapprox(G::GT, e::Identity{GT}, p; kwargs...) where {GT<:GroupManifold}
-    return isapprox(G, identity(G, p), p; kwargs...)
+isapprox(::GT, ::E, ::E; kwargs...) where {GT<:AbstractGroupManifold,E<:Identity{GT}} = true
+
+function decorator_transparent_dispatch(
+    ::typeof(isapprox),
+    M::AbstractDecoratorManifold,
+    e::E,
+    p,
+) where {E<:Identity{<:AbstractGroupManifold}}
+    return Val(:transparent)
```

Comment:
I'm not sure if we should make propagation involving `Identity` transparent. The wrapped manifold may have trouble dealing with `Identity` (that's the reason for a test failure in the basis PR).

Reply:
Actually I spent quiiiite some time getting this somehow to work, this one work here, but I don't feel good with that yet, you're right.

Reply:
So this one works with the tests here and I would propose to check this more thoroughly when anyways incorporating bases – I am not even sure how to work with identity I the bases.

Reply:
OK, since it works we can leave it for now.